pytorch_transformers/tokenization_utils.py

```diff
@@ -666,35 +690,188 @@ def _convert_token_to_id(self, token):
+                    **kwargs):
+        """
+        Returns a dictionary containing the encoded sequence or sequence pair. Other values can be returned by this
+        method: the mask for sequence classification and the overflowing elements if a ``max_length`` is specified.
+
+        Args:
+            text: The first sequence to be encoded. This can be a string, a list of strings (tokenized string using
+                the `tokenize` method) or a list of integers (tokenized string ids using the `convert_tokens_to_ids`
+                method)
+            text_pair: Optional second sequence to be encoded. This can be a string, a list of strings (tokenized
+                string using the `tokenize` method) or a list of integers (tokenized string ids using the
+                `convert_tokens_to_ids` method)
             add_special_tokens: if set to ``True``, the sequences will be encoded with the special tokens relative
                 to their model.
+            output_token_type: if set to ``True``, returns the text pair corresponding mask with 0 for the first sequence,
+                and 1 for the second.
+            max_length: if set to a number, will limit the total sequence returned so that it has a maximum length.
+                If there are overflowing tokens, those will be added to the returned dictionary
+            stride: if set to a number along with max_length, the overflowing tokens returned will contain some tokens
+                from the main sequence returned. The value of this argument defined the number of additional tokens.
+            truncate_first_sequence: if there is a specified max_length, this flag will choose which sequence
+                will be truncated.
             **kwargs: passed to the `self.tokenize()` method
         """
+
+        information = {}
+
+        def get_input_ids(text):
+            if isinstance(text, six.string_types):
+                input_ids = self.convert_tokens_to_ids(self.tokenize(text, **kwargs))
+            elif isinstance(text, (list, tuple)) and len(text) > 0 and isinstance(text[0], six.string_types):
+                input_ids = self.convert_tokens_to_ids(text)
+            elif isinstance(text, (list, tuple)) and len(text) > 0 and isinstance(text[0], int):
+                input_ids = text
+            else:
+                raise ValueError("Input is not valid. Should be a string, a list/tuple of strings or a list/tuple of integers.")
+
+            return input_ids
+
         if text_pair is None:
+            sequence_tokens = get_input_ids(text)
+
             if add_special_tokens:
-                return self.add_special_tokens_single_sentence(self.convert_tokens_to_ids(self.tokenize(text, **kwargs)))
+                information = self.prepare_for_model(sequence_tokens, max_length=max_length, stride=stride)
             else:
-                return self.convert_tokens_to_ids(self.tokenize(text, **kwargs))
+                if max_length:
```

Comment:
Shouldn't we have this `max_length` test before the `add_special_tokens` test (and one after the other instead of being in a `if..then` structure?

Reply:
Ok, see that's in the `prepare_model` methode